SteamKit2/SteamKit2/Steam/CDN/ClientLancache.cs

```diff
@@ -0,0 +1,40 @@
+﻿/*
+ * This file is subject to the terms and conditions defined in
+ * file 'license.txt', which is part of this source code package.
+ */
+
+using System;
+using System.Net.Http;
+
+namespace SteamKit2.CDN
+{
+    public partial class Client
+    {
+        /// <summary>
+        /// When set to true, will attempt to download from a Lancache instance on the LAN rather than going out to Steam's CDNs.
+        /// </summary>
+        public static bool UseLancacheServer { get; set; }
+
+        static HttpRequestMessage BuildLancacheRequest( Server server, string command, string? query)
+        {
+            var builder = new UriBuilder()
+            {
+                Scheme = "http",
+                Host = "lancache.steamcontent.com",
+                Port = 80,
+                Path = command,
+                Query = query ?? string.Empty
+            };
+
+            var request = new HttpRequestMessage( HttpMethod.Get, builder.Uri );
+            if ( UseLancacheServer )

```

Comment:
you can't get here without this being true

Reply:
Thanks, good catch.